### PR TITLE
Use filepath join

### PR DESCRIPTION
**Pull request by dhellmann**
Use filepath.Join() to combine strings to build filenames instead of relying on string concatenation. This is more reliable if the configured directory name ends in a path separator character.

**Which issue(s) this PR addresses**:

[USHIFT-180](https://issues.redhat.com//browse/USHIFT-180)

### CODE REVIEW

File: pkg/cmd/init.go [hunk 1]
```diff
@@ -141,23 +142,28 @@ func initServerConfig(cfg *config.MicroshiftConfig) error {
...
+		"kube-apiserver", []string{"kube-apiserver", "system:kube-apiserver", "system:masters"}, cfg.Cluster.URL); err != nil {
 		return err
 	}
-	if err := util.Kubeconfig(cfg.DataDir+"/resources/kube-controller-manager/kubeconfig", "system:kube-controller-manager", []string{"system:kube-controller-manager"}, cfg.Cluster.URL); err != nil {
+	if err := util.Kubeconfig(filepath.Join(cfg.DataDir, "/resources/kube-controller-manager/kubeconfig"),
+		"system:kube-controller-manager", []string{"system:kube-controller-manager"}, cfg.Cluster.URL); err != nil {
 		return err
-	}
-	if err := util.Kubeconfig(cfg.DataDir+"/resources/kube-scheduler/kubeconfig", "system:kube-scheduler", []string{"system:kube-scheduler"}, cfg.Cluster.URL); err != nil {
+	} if err := util.Kubeconfig(filepath.Join(cfg.DataDir, "/resources/kube-scheduler/kubeconfig"),
```
Comment (ggiguash) [inline pkg/cmd/init.go:156]: A nit: this probably should be on the next line.

**Review by ggiguash — APPROVED** [review]

**Comment by dhellmann** [comment]
/hold

There's no point continually testing.

File: pkg/cmd/init.go (hunk 1 above)
Comment (mangelajo) [reply pkg/cmd/init.go:156]:
```suggestion
	} 
	if err := util.Kubeconfig(filepath.Join(cfg.DataDir, "/resources/kube-scheduler/kubeconfig"),
```

**Review by mangelajo — COMMENTED** [review]
/lgtm with a little comment from @ggiguash 

CI is upside down

File: pkg/cmd/init.go (hunk 1 above)
Comment (dhellmann) [reply pkg/cmd/init.go:156]: That's so weird. This file does not look like this on my local system. There's a comment on that line instead. I've added a commit to adjust the whitespace to see if that helps with the rendering.

Comment (dhellmann) [reply pkg/cmd/init.go]: And of course I was on the wrong branch. This is fixes.

**Review by ggiguash — APPROVED** [review]

**Pull request merged**